lib/IHP/static/helpers.js

```diff
@@ -296,7 +296,15 @@ window.submitForm = function (form, possibleClickedButton) {
             'Content-Type',
             'application/x-www-form-urlencoded'
         );
-        request.send(parameters.join('&'));
+
+        try {
+            request.send(parameters.join('&'));
+        } catch (e) {
+            // If the send method throws an exception, redirect to the form action URL.
+            // This can happen for example if redirecting to an external site, which doesn't have CORS enabled,
+            // so we can't use Ajax.
+            window.location.href = url;
```

Comment:
This won't work for forms that are submitted via POST. I don't think this can be really fixed at all except by disabling the JS form submission in those forms

Reply:
Ok, I'll share my example code soon to show my use case

Reply:
Here's the example: https://github.com/amitaibu/ihp-landing-page/pull/35

If you navigate to `http://localhost:8000` without the PR it will result with console error. With the PR you'll be redirected to the repo of one of the best Haskell frameworks! :smile:

Reply:
@mpscholten ^^